### PR TITLE
SONARIAC-981 ARM Bicep add forCondition to resourceDecl

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**
Complete integration and test of `OBJECT_EXPRESSION()` into `PRIMARY_EXPRESSION()` should be done properly in `SONARIAC-999`.

### CODE REVIEW

File: iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/impl/bicep/ResourceDeclarationImplTest.java [hunk 1]
```diff
@@ -64,7 +66,7 @@ void shouldParseMinimalResourceDeclarationObject(String body) {
     assertThat(((Identifier) tree.children().get(2)).value()).isEqualTo("myName");
     assertThat(((StringComplete) tree.children().get(3)).value()).isEqualTo("type@version");
     assertThat(((SyntaxToken) tree.children().get(4)).value()).isEqualTo("=");
-    assertThat(tree.children().get(5)).isInstanceOfAny(ObjectExpression.class, IfExpression.class);
+    assertThat(tree.children().get(5)).isInstanceOfAny(ObjectExpression.class, IfExpression.class, ForExpression.class);
```
Comment (nils-werner-sonarsource) [inline iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/impl/bicep/ResourceDeclarationImplTest.java:69]: This is not a precise test. It is one of the tests which try to cover everything. When doing a parameterized test you should actually test the differences. The common things can be checked in a dedicated one.
isInstanceOfAny is not precise. 
Additionally, please do not test the elements of a tree with the `children()` method. Aren't there any getters?
Please refactor the entire test approach. I'm happy to discuss a clean approach.

File: its/ruling/src/test/resources/expected/azureresourcemanager/azureresourcemanager-S2260.json
Comment (nils-werner-sonarsource) [inline its/ruling/src/test/resources/expected/azureresourcemanager/azureresourcemanager-S2260.json:1]: This looks good!

File: iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/ResourceDeclarationImpl.java [hunk 2]
```diff
@@ -145,8 +147,15 @@ public StringLiteral type() {
   public List<Property> properties() {
     if (expression.is(Kind.OBJECT_EXPRESSION)) {
       return ((ObjectExpression) expression).properties();
-    } else {
+    } else if (expression.is(Kind.IF_EXPRESSION)) {
```
Comment (nils-werner-sonarsource) [inline iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/ResourceDeclarationImpl.java:150]: Why do we call it Expression not Condition?

**Review by nils-werner-sonarsource — CHANGES_REQUESTED** [review]
Even when you only extended the parameterized test for the ResourceDeclarationImpl please refactor the method to test more precise and dedicated.

File: iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/ResourceDeclarationImpl.java (hunk 2 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/ResourceDeclarationImpl.java]: Good point, I guess it's a mistake, [SONARIAC-1007](https://sonarsource.atlassian.net/browse/SONARIAC-1007) on the way to fix this.

File: iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/impl/bicep/ResourceDeclarationImplTest.java (hunk 1 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/arm/src/test/java/org/sonar/iac/arm/tree/impl/bicep/ResourceDeclarationImplTest.java]: I reworked the resource tests.
I discussed there is currently no getter to work with the `ObjectExpression`/`IfCondition`/`ForExpression` yet, so the test mainly focus to ensure we can work with the properties in both cases.

**Review by nils-werner-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**